spec/jekyll/configuration_spec.rb

```diff
@@ -0,0 +1,37 @@
+RSpec.describe(Jekyll::Configuration) do
+  let(:config) { described_class.from({}) }
+
+  describe "#safe_load_file" do
+    it "throws an InvalidFileFormat on a bad extension" do
+      filename = "my_config.json"
+      expect(-> { config.safe_load_file(filename) }).to raise_error(
+        Jekyll::Errors::InvalidFileFormatError,
+        "No parser for '#{filename}' is available. Use a .toml or .y(a)ml file instead."
```

Comment:
Never test the message unless you are sure it will remain static, this leads to coupled testing that is refactor-unfriendly and leads us down the slippery slope that is currently our coupled testing... with another library.